backend/services/deviceauth/server.go

```diff
@@ -160,3 +158,30 @@ func RunServer(c config.Reader) error {
 	}
 	return nil
 }
+
+func setupRedis(c config.Reader, connStr string) (cache.Cache, *rate.HTTPLimiter, error) {
+	l := log.NewEmpty()
+	l.Infof("setting up redis cache")
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
```

Comment:
will this cancel the context right after setup? do we need this context after this call?